File: app/src/main/java/com/ncapdevi/sample/fragments/FavoritesFragment.java
package com.ncapdevi.sample.fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;

/**
 * Created by niccapdevila on 3/26/16.
 */
public class FavoritesFragment extends BaseFragment {

    public static FavoritesFragment  newInstance(int instance) {
        Bundle args = new Bundle();
        args.putInt(ARGS_INSTANCE, instance);
        FavoritesFragment fragment = new FavoritesFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (btn != null) {
            btn.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (mFragmentNavigation != null) {
                        mFragmentNavigation.pushFragment(FavoritesFragment.newInstance(mInt+1));
                    }
                }
            });
            btn.setText(getClass().getSimpleName() + " " + mInt);
        }
    }
}

File: app/src/main/java/com/ncapdevi/sample/fragments/FavoritesFragment.kt
package com.ncapdevi.sample.fragments

import android.annotation.SuppressLint
import android.os.Build
import android.os.Bundle
import android.transition.TransitionInflater
import android.view.View
import android.widget.CheckBox
import android.widget.EditText
import com.ncapdevi.sample.R

/**
 * Created by niccapdevila on 3/26/16.
 */
class FavoritesFragment : BaseFragment() {

    @SuppressLint("SetTextI18n")
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                sharedElementEnterTransition = TransitionInflater.from(context).inflateTransition(
                        android.R.transition.fade
                )
            }
        }

        val et = view.findViewById<EditText>(R.id.edit_text)
        val cb = view.findViewById<CheckBox>(R.id.checkbox)

        val list = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            listOf<Pair<View, String>>(
                    Pair(et, et.transitionName),
                    Pair(cb, cb.transitionName)
            )
        } else {
            listOf()
        }

        btn.setOnClickListener {
            mFragmentNavigation.pushFragment(newInstance(mInt + 1), list)
        }
        btn.text = """${javaClass.simpleName} $mInt"""
    }

    companion object {

        fun newInstance(instance: Int): FavoritesFragment {
            val args = Bundle()
            args.putInt(BaseFragment.ARGS_INSTANCE, instance)
            val fragment = FavoritesFragment()
            fragment.arguments = args
            return fragment
        }
    }
}
